spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkTable.java

```diff
@@ -295,7 +295,7 @@ public void deleteWhere(Filter[] filters) {
           .deleteFromRowFilter(deleteExpr)
           .commit();
     } catch (ValidationException e) {
-      throw new IllegalArgumentException("Failed to cleanly delete data files matching: " + deleteExpr, e);
+      throw new ValidationException(e, "Failed to cleanly delete data files matching: %s", deleteExpr);
```

Comment:
Thanks, I mean, can we remove catch block entirely?  (It should throw the ValidationException directly from the method, no need to catch and re-throw)

Reply:
@szehon-ho just to confirm you want the `ValidationException` directly thrown by the method `deleteWhere`?
Like this:
```
@Override
  public void deleteWhere(Filter[] filters) throws ValidationException {
    Expression deleteExpr = SparkFilters.convert(filters);

    if (deleteExpr == Expressions.alwaysFalse()) {
      LOG.info("Skipping the delete operation as the condition is always false");
      return;
    }

    icebergTable.newDelete()
            .set("spark.app.id", sparkSession().sparkContext().applicationId())
            .deleteFromRowFilter(deleteExpr)
            .commit();
  }
```

Reply:
Yea that's right, I think we can even remove  it from method's throw clause, as it's a RuntimeException.

Reply:
Updated the code, however regarding "we can even remove it from method's throw clause" I am not sure to follow you, if you can explain this it would be great since I thought we actually wanted to throw a `ValidationException`.

Reply:
So I mean, we don't need the declaration 'throws ValidationException' =>

``` public void deleteWhere(Filter[] filters) { ```

We will still throw it from the internal method, but don't need to declare it as it's a RuntimeException.

Reply:
Got it, that makes sense, thank you for the explanation, removed it.